app/models/service_template.rb

```diff
@@ -377,6 +377,17 @@ def create_resource_actions(ae_endpoints)
     save!
   end
 
+  def provision_request(userid, options = {})
+    provision_workflow(userid, options).submit_request
+  end
+
+  def provision_workflow(userid, options = {})
+    ResourceActionWorkflow.new({}, User.find_by!(:userid => userid),
+                               provision_action, :target => self).tap do |wf|
+      options.each { |key, value| wf.set_value(key, value) }
```

Comment:
this might fail if method is called as provision_workflow(userid, nil)
maybe default options to nil, then have a separate options ||= {}